apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/newchat/NewChatView.kt

```diff
@@ -192,6 +207,254 @@ private fun RetryButton(onClick: () -> Unit) {
+    verticalAlignment = Alignment.CenterVertically
+  ) {
+    Row(verticalAlignment = Alignment.CenterVertically) {
+      image()
+      TextIconSpaced(false)
+      Text(title, modifier = Modifier.align(Alignment.CenterVertically))
+      if (onInfo != null) {
+        Spacer(Modifier.padding(6.dp))
+        Column(Modifier
+          .size(48.dp)
+          .clip(CircleShape)
+          .clickable(
+            enabled = !disabled,
+            onClick = { ModalManager.start.showModal { IncognitoView() } }
+          ),
+          horizontalAlignment = Alignment.CenterHorizontally,
+          verticalArrangement = Arrangement.Center
+        ) {
+          Icon(
+            painterResource(MR.images.ic_info),
+            stringResource(MR.strings.incognito),
+            tint = MaterialTheme.colors.primary
+          )
+        }
+      }
+    }
+    if (selected) {
+      Icon(painterResource(
+        MR.images.ic_check),
+        title,
+        Modifier.size(20.dp),
+        tint =  MaterialTheme.colors.primary,
+      )
+    }
+  }
+  Divider(
+    Modifier.padding(
+      start = DEFAULT_PADDING_HALF,
+      top = 4.dp,
+      end = DEFAULT_PADDING_HALF,
+      bottom = 4.dp
+    )
+  )
+}
+
+private fun filteredProfiles(users: List<User>, searchTextOrPassword: String): List<User> {
+  val s = searchTextOrPassword.trim()
+  val lower = s.lowercase()
+  return users.filter { u ->
+    if ((u.activeUser || !u.hidden) && (s == "" || u.anyNameContains(lower))) {
+      true
+    } else {
+      correctPassword(u, s)
+    }
+  }
+}
+
+@Composable
+private fun ActiveProfilePicker(
+  search: MutableState<String>,
+  contactConnection: PendingContactConnection?,
+  close: () -> Unit,
+  rhId: Long?
+) {
+  val switchingProfile = remember { mutableStateOf(false) }
+  val incognito = remember {
+    chatModel.showingInvitation.value?.conn?.incognito ?: controller.appPrefs.incognito.get()
+  }
+  val selectedProfile by remember { derivedStateOf { chatModel.currentUser.value } }
+  val searchTextOrPassword = rememberSaveable { search }
+  val profiles = remember {
+    chatModel.users.map { it.user }.sortedBy { !it.activeUser }
+  }
+  val filteredProfiles by remember {
+    derivedStateOf { filteredProfiles(profiles, searchTextOrPassword.value) }
+  }
+
+  var progressByTimeout by rememberSaveable { mutableStateOf(false) }
+
+  LaunchedEffect(switchingProfile.value) {
+    progressByTimeout = if (switchingProfile.value) {
+      delay(500)
+      switchingProfile.value
+    } else {
+      false
+    }
+  }
+
+  @Composable
+  fun profilePickerUserOption(user: User) {
+    val selected = selectedProfile?.userId == user.userId && !incognito
+
+    ProfilePickerOption(
+      title = user.chatViewName,
+      disabled = switchingProfile.value || selected,
+      selected = selected,
+      onSelected = {
+        switchingProfile.value = true
+        withApi {
+          if (contactConnection != null) {
```

Comment:
put the whole block into try { ... } finally { switchingProfile.value = false } because if something will crash, that thing will never become `false`